cmd/node/node.go

```diff
@@ -290,33 +289,34 @@ func realMain() int {
 	}
 	nc.h = logging.DefaultHandler(nc.lp)
 	slog.SetDefault(slog.New(nc.h))
-	err := run(nc)
+
+	ctx, done := signal.NotifyContext(context.Background(), os.Interrupt)
+	defer done()
+
+	err := run(ctx, nc)
 	if err != nil {
 		slog.Error("Failed to run node", logging.Error(err))
 		return 1
 	}
 	return 0
 }
 
-func run(nc *config) (retErr error) {
-	eg, ctx := errgroup.WithContext(context.Background())
+func run(ctx context.Context, nc *config) (retErr error) {
+	eg, egCtx := errgroup.WithContext(ctx)
```

Comment:
I think it's better to leave `ctx`it without changes, because old `ctx` and be accidentally used instead of the new one.

Reply:
I think not, `egCtx` intended to be used only in `eg`'s goroutines.
Perviously, the `eg`'s context was derived from the background context and overwritten by the context of `NotifyContext` derived from `eg`'s context. Now, the "notify" context comes from outside and `egCtx` is used only in controlled goroutines, `runNode` and prometheus started under the "notify" context because they are not controlled by the error group.